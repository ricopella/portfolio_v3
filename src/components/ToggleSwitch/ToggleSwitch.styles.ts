import styled from '@emotion/styled'
import { BREAKPOINTS, COLORS } from '../../styles/variables'

const ToggleSwitchContainer = styled('div')`
  display: grid;
  align-items: center;
  grid-template-columns: repeat(2, max-content);
  justify-content: center;
  width: 100%;
`

const ToggleButton = styled('button')`
  background-color: transparent;
  border-radius: 3.4375rem;
  border: 0.25rem solid ${COLORS.terminalBackground};
  color: ${COLORS.terminalBackground};
  font-size: 1.25rem;
  font-weight: bold;
  height: 2.5rem;
  text-transform: capitalize;
  width: 9.375rem;
  letter-spacing: 0.1rem;

  &.active {
    background-color: ${COLORS.terminalBackground};
    color: ${COLORS.white};

    &:hover {
      cursor: not-allowed;
    }
  }

  &.left {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }

  &.right {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }

  &:hover {
    cursor: pointer;
  }

  @media (max-width: ${BREAKPOINTS.smRem}) {
    font-size: 1rem;
    height: 2.5rem;
    width: 7.8125rem;
  }
`

export default {
  ToggleSwitchContainer,
  ToggleButton,
}
