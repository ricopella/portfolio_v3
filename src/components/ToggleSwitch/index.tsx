import React, { FC } from 'react'
import { ToggleSwitchProps } from '../../types'
import Styled from './ToggleSwitch.styles'

/**
 * Component with two buttons that toggle one or the other
 */
const ToggleSwitch: FC<ToggleSwitchProps<string>> = ({
  currentView,
  setView,
  views,
}) => (
  <Styled.ToggleSwitchContainer>
    {views.map((view, idx) => (
      <Styled.ToggleButton
        key={`toggle_switch_${view}`}
        onClick={() => setView(view)}
        className={`${currentView === view ? 'active' : ''} ${
          idx === 0 ? 'left' : 'right'
        }`}
      >
        {view}
      </Styled.ToggleButton>
    ))}
  </Styled.ToggleSwitchContainer>
)

export default ToggleSwitch
