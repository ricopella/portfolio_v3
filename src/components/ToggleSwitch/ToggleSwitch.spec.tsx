import renderer from 'react-test-renderer'
import React from 'react'
import ToggleSwitch from './index'
import { render, fireEvent } from '@testing-library/react'

describe('<ToggleSwitch />', () => {
  const props = {
    currentView: 'foo',
    views: ['foo', 'bar'],
    setView: () => null,
  }
  it('should match ToggleSwitch snapshot', () => {
    const internalTree = renderer.create(<ToggleSwitch {...props} />).toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should show the views provided', () => {
    const { getByText } = render(<ToggleSwitch {...props} />)

    expect(getByText(/foo/i)).toBeTruthy()
    expect(getByText(/bar/i)).toBeTruthy()
  })

  it('should handle clicking the toggle', () => {
    const spy = jest.fn()
    const { getByText } = render(<ToggleSwitch {...props} setView={spy} />)

    fireEvent.click(getByText(/bar/i))

    expect(spy).toHaveBeenCalledTimes(1)
  })
})
