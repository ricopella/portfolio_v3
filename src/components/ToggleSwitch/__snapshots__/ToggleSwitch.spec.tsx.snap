// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> should match ToggleSwitch snapshot 1`] = `
<div
  className="css-1bapo22"
>
  <button
    className="active left css-msm5so"
    onClick={[Function]}
  >
    foo
  </button>
  <button
    className=" right css-msm5so"
    onClick={[Function]}
  >
    bar
  </button>
</div>
`;
